src/goto-analyzer/goto_analyzer_parse_options.cpp

```diff
@@ -304,40 +577,120 @@ int goto_analyzer_parse_optionst::doit()
       std::cout << '\n';
     }
 
-    return 0;
+    return CPROVER_EXIT_SUCCESS;
   }
 
   label_properties(goto_model);
 
   if(cmdline.isset("show-properties"))
   {
     show_properties(goto_model, get_ui());
-    return 0;
+    return CPROVER_EXIT_SUCCESS;
   }
 
   if(set_properties())
-    return 7;
+    return CPROVER_EXIT_SET_PROPERTIES_FAILED;
 
-  if(cmdline.isset("show-intervals"))
+  if(options.get_bool_option("general-analysis"))
   {
-    show_intervals(goto_model, std::cout);
-    return 0;
-  }
 
-  if(cmdline.isset("non-null") ||
-     cmdline.isset("intervals"))
-  {
-    optionst options;
-    options.set_option("json", cmdline.get_value("json"));
-    options.set_option("xml", cmdline.get_value("xml"));
-    bool result=
-      static_analyzer(goto_model, options, get_message_handler());
-    return result?10:0;
+    // Output file factory
+    const std::string outfile=options.get_option("outfile");
+    std::ofstream output_stream;
+    if(!(outfile=="-"))
+      output_stream.open(outfile);
+
+    std::ostream &out((outfile=="-") ? std::cout : output_stream);
```

Comment:
Is it cleaner/possible to factor together lines 599-603 here, and do something like:
```
std::ostream &out((outfile=="-") ? std::cout : std::ofstream(outfile));
```
?

Reply:
I tried that and C++ complained.  I think the rule is the you aren't allowed to use an rvalue to initialise a non-constant reference.